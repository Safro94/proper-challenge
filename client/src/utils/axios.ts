import axios, { AxiosError, AxiosRequestConfig } from 'axios';
import https from 'https';

const opts: AxiosRequestConfig = {
	httpsAgent: new https.Agent({ rejectUnauthorized: false }),
};

const interceptorError = (error: AxiosError) => Promise.reject(error.response);

const instance = axios.create(opts);

instance.interceptors.response.use(response => response, interceptorError);
instance.defaults.headers.common['x-api-key'] = process.env.REACT_APP_API_KEY;

export default instance;
