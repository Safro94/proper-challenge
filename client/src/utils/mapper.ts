import { IAddress, IDawaResponse } from '../types';

const mapDawaResponse = (data: IDawaResponse[]): IAddress[] => {
	return data.map(item => ({
		text: item.tekst,
		type: item.type,
	}));
};

export { mapDawaResponse };
