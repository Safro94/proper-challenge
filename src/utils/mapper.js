const mapProperties = properties =>
	properties.map(({ address, size, rooms, utilities, tenantName }) => ({
		address,
		size,
		rooms,
		utilities,
		tenantName,
	}));

module.exports = { mapProperties };
