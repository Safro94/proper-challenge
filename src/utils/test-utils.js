const mockReq = params => {
	const req = { ...params };
	return req;
};

const mockRes = params => {
	const res = { ...params };
	res.json = jest.fn().mockReturnValue(res);
	res.status = jest.fn().mockReturnValue(res);

	return res;
};

module.exports = {
	mockReq,
	mockRes,
};
