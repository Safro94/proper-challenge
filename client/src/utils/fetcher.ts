import { RequestMethods, RequestStatus } from '../types';

import axios from './axios';

interface IFetcher {
	method?: RequestMethods;
	url: string;
	data?: null | any;
	body?: null | any;
	headers?: null | any;
}

interface IResponse {
	data: any;
	status: RequestStatus;
}

const fetcher = async ({
	method = RequestMethods.GET,
	url,
	body = null,
	headers = null,
}: IFetcher): Promise<IResponse> => {
	let data;
	let status = RequestStatus.Pending;

	try {
		const result = await axios[method](
			url,
			JSON.parse(headers),
			JSON.parse(body)
		);
		data = result.data;
		status = RequestStatus.Resolved;
	} catch (error) {
		status = RequestStatus.Rejected;
		throw error;
	}

	return { data, status };
};

export default fetcher;
