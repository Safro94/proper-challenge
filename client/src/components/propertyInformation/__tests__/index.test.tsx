import { render, screen } from '../../../utils/test-utils';

import PropertyInformation from '..';

describe('PropertyInformation', () => {
	const property = {
		tenantName: 'Matias',
		rooms: 3,
		size: '40mt2',
		address: 'abc',
		utilities: 'def',
	};

	it('should render correctly', () => {
		const { container } = render(
			<PropertyInformation property={{ ...property, id: 1 }} />
		);
		expect(container.firstChild).toMatchSnapshot();
	});

	it('should render 5 items', () => {
		const texts = ['tenant', 'rooms', 'size', 'address', 'utilities'];

		render(<PropertyInformation property={{ ...property, id: 1 }} />);

		texts.forEach(text => {
			expect(screen.getByText(text)).toBeInTheDocument();
		});

		Object.values(property).forEach(item => {
			expect(screen.getByText(item)).toBeInTheDocument();
		});
	});
});
