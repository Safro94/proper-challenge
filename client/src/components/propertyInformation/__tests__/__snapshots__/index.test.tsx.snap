// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyInformation should render correctly 1`] = `
<div
  class="sc-bdnxRM bDiZob"
>
  <div
    class="sc-gtsrHT JXsyT"
  >
    <h4
      class="sc-dlnjwi"
    >
      tenant
    </h4>
    <h5
      class="sc-hKFxyN cLDGBv"
    >
      Matias
    </h5>
  </div>
  <div
    class="sc-gtsrHT JXsyT"
  >
    <h4
      class="sc-dlnjwi"
    >
      rooms
    </h4>
    <h5
      class="sc-hKFxyN cLDGBv"
    >
      3
    </h5>
  </div>
  <div
    class="sc-gtsrHT JXsyT"
  >
    <h4
      class="sc-dlnjwi"
    >
      size
    </h4>
    <h5
      class="sc-hKFxyN cLDGBv"
    >
      40mt2
    </h5>
  </div>
  <div
    class="sc-gtsrHT JXsyT"
  >
    <h4
      class="sc-dlnjwi"
    >
      address
    </h4>
    <h5
      class="sc-hKFxyN cLDGBv"
    >
      abc
    </h5>
  </div>
  <div
    class="sc-gtsrHT JXsyT"
  >
    <h4
      class="sc-dlnjwi"
    >
      utilities
    </h4>
    <h5
      class="sc-hKFxyN cLDGBv"
    >
      def
    </h5>
  </div>
</div>
`;
